librosa/feature/spectral.py

```diff
@@ -1674,7 +1682,14 @@ def mfcc(y=None, sr=22050, S=None, n_mfcc=20, dct_type=2, norm='ortho', **kwargs
     if S is None:
         S = power_to_db(melspectrogram(y=y, sr=sr, **kwargs))
 
-    return scipy.fftpack.dct(S, axis=0, type=dct_type, norm=norm)[:n_mfcc]
+    M = scipy.fftpack.dct(S, axis=0, type=dct_type, norm=norm)[:n_mfcc]
+
+    if lifter > 0:
+        M *= 1 + (lifter / 2) * np.sin(np.pi * np.arange(1, 1 + n_mfcc) / lifter)[:, np.newaxis]
```

Comment:
if `lifter` is very large, `M` is unchanged.
but if `lifter` overflows, `M` is changed to a constant array of `np.inf` values.

If would be in favor of turning this condition into `0 < lifter < np.inf`

Reply:
I don't see a great need to explicitly handle infiities here.  I can't imagine someone accidentally using an infinite lifter coefficient, and if they do it on purpose, then it's on them.

Reply:
OK, here's what i propose. Turn the `np.sin` into a sine cardinal: `np.sinc`. 
The infinity case will nicely fall back to `np.sinc(0) == 1.0`, and the result will be well-defined.
It will also be more numerically stable for large finite values of the `lifter/n_mfcc` ratio.

Reply:
Also, please don't forget the `dtype=y.dtype` in `np.arange`

Reply:
+1 for dtype.

Re sinc: the lifter isn't really a sinc, so the equivalent filter looks like 

`1 + np.sinc(np.arange(1, N+1) / L) * np.arange(1, N+1) * np.pi / 2`

Comparing the two using `float32` and `lifter=1e80` still agree down to 6 decimal places, so I'm not sure there's much benefit here.